src/openeo_pg_parser/utils.py

```diff
@@ -0,0 +1,198 @@
+        elif isinstance(src, str) and url_is_valid(src):
+            r = requests.get(url=src)
+            data = r.json()
+            process_list = data['processes']
+        elif isinstance(src, list):
+            process_list = src
+        else:
+            err_msg = "Either a processes URL or a local directory path must be specified."
+            raise ValueError(err_msg)
+
+        processes = {}
+        for process in process_list:
+            processes[process['id']] = process
+
+    return processes
+
+
+def load_collections(src):
+    """
+    Collects collection definitions from a local process directory, from a URL or a list of collection
+    definitions.
+
+    Parameters
+    ----------
+    src : dict or str or list, optional
+            It can be:
+                - dictionary of loaded collection definitions (keys are the collection ID's)
+                - directory path to collections (.json)
+                - URL of the remote collection endpoint (e.g., "https://earthengine.openeo.org/v1.0/collections")
+                - list of loaded collection definitions
+
+    Returns
+    -------
+    dict :
+        Dictionary linking collection IDs with the respective collection definitions.
+
+    """
+
+    if isinstance(src, dict):
+        collections = src
+    else:
+        if isinstance(src, str) and os.path.isdir(src):
+            filepaths = glob.glob(os.path.join(src, "*.json"))
+            collection_list = [load_json_file(filepath) for filepath in filepaths]
+        elif isinstance(src, str) and url_is_valid(src):
+            r = requests.get(url=src)
+            data = r.json()
+            collection_list = data['collections']
+        elif isinstance(src, list):
+            collection_list = src
+        else:
+            err_msg = "Either a collections URL or a local directory path must be specified."
+            raise ValueError(err_msg)
+
+        collections = {}
+        for collection in collection_list:
+            collections[collection['id']] = collection
+
+    return collections
+
+
+def keys2str(keys):
+    """
+    Converts list of keys to a string, which can be used for indexing (e.g., a dictionary).
+
+    Parameters
+    ----------
+    keys : list of str
+        List of keys.
+
+    Returns
+    -------
+    str
+        Indexing string.
+    """
+
+    key_str = ""
+    for key in keys:
+        if isinstance(key, str):
+            key = "'{}'".format(key)
+        key_str += "[{}]".format(key)
+
+    return key_str
+
+def get_obj_elem_from_keys(obj, keys):
+    """
+    Returns values stored in `obj` by using a list of keys for indexing.
+
+    Parameters
+    ----------
+    obj : object
+        Python object offering indexing, e.g., a dictionary.
+    keys : list of str
+        List of keys for indexing.
+
+    Returns
+    -------
+    object
+        Values of the indexed object.
+    """
+
+    key_str = keys2str(keys)
+    return copy.deepcopy(eval('obj' + key_str))
```

Comment:
this trick with `eval` and building a long indexing string looks bad practice to me
why not just use recursion? e.g.

```python
def get_obj_elem_from_keys(obj, keys):
    if len(keys) > 1:
        return get_obj_elem_from_keys(obj[keys[0]], keys[1:])
    else:
        return obj[keys[0]]
```

Reply:
I agree, thats much better and cleaner, thanks!